stdlib/string.ml

```diff
@@ -280,3 +280,98 @@ let get_int32_le s i = B.get_int32_le (bos s) i
 let get_int32_be s i = B.get_int32_be (bos s) i
 let get_int64_le s i = B.get_int64_le (bos s) i
 let get_int64_be s i = B.get_int64_be (bos s) i
+
+(* Spellchecking *)
+
+let uchar_utf_8_byte_decode_length = function
+  | '\x00' .. '\x7F' -> 1
+  | '\x80' .. '\xC1' -> 0
+  | '\xC2' .. '\xDF' -> 2
+  | '\xE0' .. '\xEF' -> 3
+  | '\xF0' .. '\xF4' -> 4
+  | _ -> 0
+
+let utf_8_uchar_length s =
+  let slen = length s in
+  let i = ref 0 and ulen = ref 0 in
+  while (!i < slen) do
+    let dec_len = uchar_utf_8_byte_decode_length (unsafe_get s !i) in
+    i := (!i + if dec_len = 0 then 1 (* count one Uchar.rep *) else dec_len);
+    incr ulen;
+  done;
+  !ulen
+
+let uchar_array_of_utf_8_string s =
+  let slen = length s in (* is an upper bound on Uchar.t count *)
+  let uchars = Array.make slen Uchar.max in
+  let k = ref 0 and i = ref 0 in
+  while (!i < slen) do
+    let dec = get_utf_8_uchar s !i in
+    i := !i + Uchar.utf_decode_length dec;
+    uchars.(!k) <- Uchar.utf_decode_uchar dec;
+    incr k;
+  done;
+  uchars, !k
+
+let _edit_distance ?(limit = Int.max_int) s (s0, len0) s1 =
```

Comment:
Not asking for a change, but I would have avoided the `_` here which is normally used to identify unused functions.